src/main/java/one/util/streamex/MoreCollectors.java

```diff
@@ -168,6 +168,209 @@ private MoreCollectors() {
+     *                    values associated with the same key, as supplied
+     *                    to {@link Map#merge(Object, Object, BiFunction)}
+     * @param mapSupplier a function which returns a new, empty {@code Map} into
+     *                    which the results will be inserted
+     * @return {@code Collector} which collects elements into a {@code Map}
+     * whose keys and values are taken from {@code Map.Entry} and combining them
+     * using the {@code combiner} function
+     *
+     * @see Collectors#toMap(Function, Function, BinaryOperator, Supplier)
+     */
+    public static <K, V, M extends Map<K, V>> Collector<Entry<K, V>, ?, M> entriesToCustomMap(
+            BinaryOperator<V> combiner, Supplier<M> mapSupplier) {
+        return Collectors.toMap(Entry::getKey, Entry::getValue, combiner, mapSupplier);
+    }
+
+    /**
+     * Returns a {@code Collector} that accumulates elements into
+     * a result {@code Map} defined by {@code mapSupplier} function
+     * whose keys are taken from {@code Map.Entry} and values are the result
+     * of applying the provided {@code valueMapper} function to {@code Map.Entry} values
+     * and combining them using the provided {@code combiner} function.
+     *
+     * <p>If the mapped keys contains duplicates (according to {@link Object#equals(Object)}),
+     * the value mapping function is applied to each equal element, and the
+     * results are merged using the provided {@code combiner} function.
+     *
+     * @param <K>         the {@link Comparable} type of then map keys
+     * @param <V>         the type of the map values
+     * @param <VV>        the output type of the value mapping function
+     * @param <M>         the type of the resulting {@code Map}
+     * @param valueMapper a mapping function to produce values
+     * @param combiner    a merge function, used to resolve collisions between
+     *                    values associated with the same key, as supplied
+     *                    to {@link Map#merge(Object, Object, BiFunction)}
+     * @param mapSupplier a function which returns a new, empty {@code Map} into
+     *                    which the results will be inserted
+     * @return {@code Collector} which collects elements into a {@code Map}
+     * whose keys and values are taken from {@code Map.Entry} and combining them
+     * using the {@code combiner} function
+     *
+     * @see Collectors#toMap(Function, Function, BinaryOperator, Supplier)
+     */
+    public static <K, V, VV, M extends Map<K, VV>> Collector<Entry<K, V>, ?, M> entriesToCustomMap(
+            Function<V, VV> valueMapper, BinaryOperator<VV> combiner, Supplier<M> mapSupplier) {
+        Objects.requireNonNull(valueMapper);
+        return Collectors.toMap(Entry::getKey, entry -> valueMapper.apply(entry.getValue()), combiner, mapSupplier);
+    }
+
+    private static <T> BinaryOperator<T> throwingMerger() {
```

Comment:
throwingMerger doesn't report the key for duplicate values (its parameters are actually values, not keys, so the message is misleading). This was fixed in Java 9, and I prefer to mock this behavior in StreamEx as well. See `one.util.streamex.EntryStream#toCustomMap(java.util.function.Supplier<M>)` implementation. Likely `one.util.streamex.AbstractStreamEx#addToMap` could be reused here if we don't delegate to `Collectors.toMap` but create a collector from the scratch instead using `Collector.of`.